src/xref2/component.ml

```diff
@@ -578,251 +598,434 @@ module Fmt = struct
     | [] -> ()
     | t -> fpf ppf fmt_outer (Format.pp_print_list ~pp_sep pp_a) t
 
-  let rec signature ppf sg =
+  let wrap :
+      type a.
+      config ->
+      string ->
+      (config -> Format.formatter -> a -> unit) ->
+      Format.formatter ->
+      a ->
+      unit =
+   fun c txt fn ppf x ->
+    if c.short_paths then Format.fprintf ppf "%a" (fn c) x
+    else Format.fprintf ppf "%s(%a)" txt (fn c) x
+
+  let wrap2 :
+      type a b.
+      config ->
+      string ->
+      (config -> Format.formatter -> a -> unit) ->
+      (config -> Format.formatter -> b -> unit) ->
+      Format.formatter ->
+      a ->
+      b ->
+      unit =
+   fun c txt fn1 fn2 ppf x y ->
+    if c.short_paths then Format.fprintf ppf "%a" (fn1 c) x
+    else Format.fprintf ppf "%s(%a,%a)" txt (fn1 c) x (fn2 c) y
+
+  let wrap2r :
```

Comment:
I'm a bit worried about the added maintenance in this module. Component changes often and the printing functions are slowing down other work.
These functions need some documentation.

Reply:
That's true, I'll add some doc comments.